Create Frontend Template App (#80)
AbstractSDK/templates: frontend/README.md

# Abstract Account App

## Introduction

> The Abstract Account App is a Next.js-based application that interacts with the
> Abstract Account system on the blockchain. It enables users to connect their
> wallets, create and manage Abstract Accounts, query account information, and
> interact with smart contracts in a type-safe manner using Abstract Codegen.

Reviewer: ```suggestion
This Abstract App Template App is a Next.js-based application that demonstrates
interaction with Abstract Accounts on a CosmWasm-enabled chain. It includes logic for wallet connection, Account creation & queries, and shows interaction with general CosmWasm smart-contracts using Abstract.js codegen.
```

> 

Reviewer: We should include a techstack connection 
react
typescript 
Next.js 
shadcn
tailwind

graz
react-query
abstract.js

with links to their documentation.

## Features

- Wallet connection and management using Graz
- Creation of Abstract Accounts

> - Querying Abstract Account information via the Abstract Subgraph GraphQL API

Reviewer: ```suggestion
- Querying Abstract Account information via the Abstract GraphQL API
```

> - Code generation for type-safe interactions with smart contracts using Abstract
>   Codegen

Reviewer: ```suggestion
- Code generation for type-safe interactions with smart contracts using [Abstract.js](https://abstract.js.abstract.money)
```

## Prerequisites

Before you begin, ensure you have met the following requirements:

- pnpm

> - A compatible wallet (e.g., Keplr for Cosmos-based chains)

Reviewer: ```suggestion
- A Cosmos-compatible wallet (e.g., [Keplr](keplr.app))
```

## Installation

1. Clone the repository:
   ```
   git clone git@github.com:AbstractSDK/templates.git
   ```

2. Navigate to the project directory:
   ```
   cd templates/frontend
   ```

3. Install the dependencies:
   ```
   pnpm install
   ```

## Configuration

1. Create a `.env` file in the root directory and add the following:
   ```
   ABSTRACT_SUBGRAPH_URL=https://api.abstract.money/graphql

>    ```

Reviewer: Should show duplicating the `.env.example` file

## Code Generation

This project uses code generation for type-safe interactions with smart
contracts and GraphQL queries. To generate the necessary code, run:

```
pnpm generate
```

This command will execute the following scripts:

1. `generate:graz`: Generates chain information for Graz
2. `generate:gql`: Generates TypeScript types for GraphQL queries using the
   GraphQL Code Generator
3. `generate:abstract`: Generates TypeScript types, methods, and hooks for smart
   contract interactions using Abstract CLI

> 

Reviewer: We should add a section on updating `abstract.config.js` with the proper module name, namespace, and version, and note that the codegen command referencing module-generated code *must* be run after the module schema generation.

> ## Usage

Reviewer: ```suggestion
## Development
```

To start the development server, run:

```
pnpm dev
```

Navigate to `http://localhost:3000` in your web browser to use the application.

> ## Key Components
> 
> ### WalletConnection
> 
> Handles wallet connection using Graz.
> 
> ### CreateAbstractAccount
> 
> Allows users to create new Abstract Accounts.
> 
> ### QueryAbstractSubgraph
> 
> Queries and displays Abstract Account information using Abstract Subgraph
> GraphQL API.
> 
> ### CodegenContract
> 
> Demonstrates interaction with a CW20 token contract using generated code.

Reviewer: ```suggestion
```

duplicate content

## Configuration

1. Duplicate the `.env.example` file and rename it to `.env`:
   ```
   cp .env.example .env
   ```

2. Open the `.env` file and ensure it contains the following:
   ```
   ABSTRACT_SUBGRAPH_URL=https://api.abstract.money/graphql
   ```

   You can modify the values if needed.

## Updating abstract.config.ts

Before running the code generation, follow these steps:

1. Ensure that you have generated the module schema (see the
   [Rust README](https://github.com/AbstractSDK/templates/blob/main/README.md)).
   This step is crucial as it creates the necessary schema files for the code
   generation process.

>    a. Verify that the schema file inside the path specified in step 2b is named
>    `module-schema.json`.

Reviewer: ```suggestion
```

Unnecessary step
